### DISCUSSION TITLE
No more writable volumes!

### DISCUSSION THREAD

**Discussion by ginkel** [seaweedfs/seaweedfs#4649] [Q&A]
Hi there,

I am using seaweedfs-s3 (plus the other associated daemons) to archive files into time-based buckets (named `bucket-YYYYDDMM` as an example) in near-realtime. Just a few moments ago the archiving process started to fail as `seaweedfs-s3` started to produce errors:

```
E0709 14:46:40.370822 s3api_object_handlers.go:513 upload to filer error: rpc error: code = Unknown desc = failed to find writable volumes for collection:bucket-20230710 replication:100 ttl: error: No more writable volumes!
```

The cluster is comprised of three nodes each hosting a master, filer, volume and s3 process. Free disk space is currently around 200 GB each.

Is my assumption correct that unused space from older buckets (that are no longer being written to) is preventing seaweedfs from allocating more volumes? Is there a way to reclaim that space?

`-volumeSizeLimitMB` was previously set to 512. I have just reduced it to 128, which seems to have remedied the issue for now, but according to my understanding this will only affect newly created volumes.

What is a sensible rule of thumb to determine `-volumeSizeLimitMB`? My buckets contain many small files (a few kB in size each) and are rotated on a daily basis. Old buckets are no longer being written to.

Thanks,
Thilo

**Comment by chrislusf** [comment]
You can also reduce the number of volumes used per bucket/collection by configuration `volume_growth` in `master.toml`.

Keep the `total "live" buckets` * `volume_growth factor` * `volumeSizeLimitMB` less than the total available disk size.

**Comment by ginkel** [comment]
Hello again,

today my SeaweedFS cluster ran out of free volumes again. Is there a metric I could use to monitor available volumes so that I can catch the condition before it actually happens?

Thanks,
Thilo

**Reply by chrislusf** [reply]
What is the output of "volume.list" ?

**Reply by ginkel** [reply]
Unfortunately, I didn't preserve the `volume.list` output before reducing the volume size. The current output isn't of much help, is it?

I was rather thinking about a Prometheus metric that I could craft an alert rule around (but I didn't find one - maybe I missed it?).

Edit: Current output at: https://gist.github.com/ginkel/26ca0eaa7a697189bb0037de8a06ba08
